### ISSUE TITLE
Add support for WITH clause (enhancement for syntax highlighter)

### ISSUE THREAD

**Issue by stevenwdv** [phpmyadmin/sql-parser#331]
Currently the SQL syntax highlighter marks the `WITH` keyword as `Unrecognized statement type`.
MariaDB supports a `WITH` clause in combination with a `SELECT` statement since version 10.2.1 (stable=10.2.8), see https://mariadb.com/kb/en/the-mariadb-library/with/

**phpMyAdmin version:** 4.7.4

Sort of relates to phpmyadmin/phpmyadmin#13615 (but not really)

**Comment by williamdes** [comment]
Related to #165

**Comment by sturtevant** [comment]
I believe this issue is also disrupting the way phpMyAdmin result pagination works. Any query using a CTE ("WITH" clause) will return all results to the browser (even while the interface says "Showing rows 0 - 24"). This typically results in an HTTP timeout for large tables / queries.

To reproduce (any table of sufficient length will do):
```
SELECT id FROM mytable; -- pagination works for this query

WITH my_cte AS ( SELECT id FROM mytable ) SELECT id FROM my_cte; -- pagination doesn't work (all rows returned)
```

**Comment by soyuka** [comment]
Can we close this one as duplicate from https://github.com/phpmyadmin/sql-parser/issues/165 ?

**Comment by williamdes** [comment]
> Can we close this one as duplicate from #165 ?

Now that a PR is done I would say this one can stay open

**Comment by williamdes** [comment]
This will be part of 5.5.0
Implemented in: #334
